velox/dwio/dwrf/test/WriterTests.cpp

```diff
@@ -79,8 +81,114 @@ class WriterTest : public Test {
+ public:
+  SupportedCompressionTest() : supportedCompressionKind_(GetParam()) {}
+
+  static std::vector<CompressionKind> getTestParams() {
+    static std::vector<CompressionKind> params = {
+        CompressionKind::CompressionKind_NONE,
+        CompressionKind::CompressionKind_ZLIB,
+        CompressionKind::CompressionKind_ZSTD};
+    return params;
+  }
+
+ protected:
+  CompressionKind supportedCompressionKind_;
+};
+
+class AllWriterCompressionTest
+    : public WriterTest,
+      public testing::WithParamInterface<CompressionKind> {
+ public:
+  AllWriterCompressionTest() : compressionKind_(GetParam()) {}
+
+  static std::vector<CompressionKind> getTestParams() {
+    static std::vector<CompressionKind> params = {
+        CompressionKind::CompressionKind_NONE,
+        CompressionKind::CompressionKind_ZLIB,
+        CompressionKind::CompressionKind_SNAPPY,
+        CompressionKind::CompressionKind_LZO,
+        CompressionKind::CompressionKind_ZSTD,
+        CompressionKind::CompressionKind_LZ4,
+        CompressionKind::CompressionKind_GZIP,
+        CompressionKind::CompressionKind_MAX};
+    return params;
+  }
+
+ protected:
+  CompressionKind compressionKind_;
+};
+
+TEST_P(AllWriterCompressionTest, compression) {
+  std::map<std::string, std::string> overrideConfigs;
+  overrideConfigs.emplace(
+      Config::COMPRESSION.configKey(), std::to_string(compressionKind_));
+  auto config = Config::fromMap(overrideConfigs);
+  auto& writer = createWriter(config);
+  auto& context = getContext();
+  std::array<char, 10> data;
+  std::memset(data.data(), 'a', 10);
+  auto& writerSink = writer.getSink();
+
+  for (size_t i = 0; i < 3; ++i) {
+    writerSink.setMode(WriterSink::Mode::Index);
+    writerSink.addBuffer(*pool_, data.data(), data.size());
+    writerSink.setMode(WriterSink::Mode::Data);
+    writerSink.addBuffer(*pool_, data.data(), data.size());
+    writerSink.setMode(WriterSink::Mode::Footer);
+    writerSink.addBuffer(*pool_, data.data(), data.size());
+    writerSink.setMode(WriterSink::Mode::None);
+    context.stripeRowCount = 123;
+    context.stripeRawSize = 345;
+    addStripeInfo();
+    context.nextStripe();
+  }
+
+  std::string typeStr{"struct<a:int,b:float,c:string>"};
+  HiveTypeParser parser;
+  auto schema = parser.parse(typeStr);
+
+  for (size_t i = 0; i < 2; ++i) {
+    writer.addUserMetadata(
+        folly::to<std::string>(i), folly::to<std::string>(i + 1));
+  }
+  for (size_t i = 0; i < 4; ++i) {
+    getFooter().add_statistics();
+  }
+
+  if (compressionKind_ == CompressionKind::CompressionKind_SNAPPY ||
+      compressionKind_ == CompressionKind::CompressionKind_LZO ||
+      compressionKind_ == CompressionKind::CompressionKind_LZ4 ||
+      compressionKind_ == CompressionKind::CompressionKind_GZIP ||
+      compressionKind_ == CompressionKind::CompressionKind_MAX) {
+    VELOX_ASSERT_THROW(
+        writeFooter(*schema), "Unsupported dwrf compression type");
```

Comment:
Not sure if we could log out the actual compression type in exception error message? Thanks!